### ISSUE TITLE
HTTP Error 400 Bad Request when reading from AWS S3 PreSigned URL

### ISSUE THREAD

**Issue by jgcmarins** [chezou/tabula-py#312]
# Summary of your issue
I am trying to read a pdf from AWS S3 PreSigned URL and I experiencing the following error:

```
Traceback (most recent call last):
  File "main.py", line 3, in <module>
    df = tabula.read_pdf("https://my-bucket-name.s3.amazonaws.com/v1/user/62e970a844d091d90069ab7d/file/62ed00ce74b8a716407975d6?SIGNED_HASH")[0]
  File "/home/alunix/code/ddc-reader-python/ddc-reader/lib/python3.8/site-packages/tabula/io.py", line 311, in read_pdf
    path, temporary = localize_file(input_path, user_agent)
  File "/home/alunix/code/ddc-reader-python/ddc-reader/lib/python3.8/site-packages/tabula/file_util.py", line 48, in localize_file
    req = urlopen(path_or_buffer)
  File "/usr/lib/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/usr/lib/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.8/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 400: Bad Request
```

# Check list before submit

- [x] Did you read [FAQ](https://tabula-py.readthedocs.io/en/latest/faq.html)?

- [x] (Optional, but really helpful) Your PDF URL: it's a AWS S3 PreSigned URL, sorry, can't share, but the PreSigned is working because I am able to access the PDF through the browser

- [x] Paste the output of `import tabula; tabula.environment_info()` on Python REPL:

```
Python version:
    3.8.10 (default, Jun 22 2022, 20:18:18)
[GCC 9.4.0]
Java version:
    openjdk version "1.8.0_342"
OpenJDK Runtime Environment (build 1.8.0_342-8u342-b07-0ubuntu1~20.04-b07)
OpenJDK 64-Bit Server VM (build 25.342-b07, mixed mode)
tabula-py version: 2.4.0
platform: Linux-5.10.16.3-microsoft-standard-WSL2-x86_64-with-glibc2.29
uname:
    uname_result(system='Linux', node='DESKTOP-KVLP8PC', release='5.10.16.3-microsoft-standard-WSL2', version='#1 SMP Fri Apr 2 22:23:49 UTC 2021', machine='x86_64', processor='x86_64')
linux_distribution: ('Ubuntu', '20.04', 'focal')
mac_ver: ('', ('', '', ''), '')
None
```

- [x] Paste the output of `python --version` command on your terminal: Python 3.8.10
- [x] Paste the output of `java -version` command on your terminal:
```
openjdk version "1.8.0_342"
OpenJDK Runtime Environment (build 1.8.0_342-8u342-b07-0ubuntu1~20.04-b07)
OpenJDK 64-Bit Server VM (build 25.342-b07, mixed mode)
```
- [x] Does `java -h` command work well?; Ensure your java command is included in `PATH`
- [x] Write your OS and it's version: Windows 10 with WSL

# What did you do when you faced the problem?

Tried to search on both Google and GitHub issues to see if anyone else is facing the same issue and found nothing.

## Code:

```python
df = tabula.read_pdf("https://my-bucket-name.s3.amazonaws.com/v1/user/62e970a844d091d90069ab7d/file/62ed00ce74b8a716407975d6?SIGNED_HASH")[0]
df.to_csv('./test.csv', encoding='utf-8')
print(df)
```

## Expected behavior:

CSV with PDF data.

## Actual behavior:

```
Traceback (most recent call last):
  File "main.py", line 3, in <module>
    df = tabula.read_pdf("https://my-bucket-name.s3.amazonaws.com/v1/user/62e970a844d091d90069ab7d/file/62ed00ce74b8a716407975d6?SIGNED_HASH")[0]
  File "/home/alunix/code/ddc-reader-python/ddc-reader/lib/python3.8/site-packages/tabula/io.py", line 311, in read_pdf
    path, temporary = localize_file(input_path, user_agent)
  File "/home/alunix/code/ddc-reader-python/ddc-reader/lib/python3.8/site-packages/tabula/file_util.py", line 48, in localize_file
    req = urlopen(path_or_buffer)
  File "/usr/lib/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/usr/lib/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.8/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 400: Bad Request
```

**Comment by chezou** [comment]
Added `use_raw_url` option for the case to avoid the Presigned URL. Can you try it once you have a time?

**Comment by chezou** [comment]
@jgcmarins Just confirming, did you reinstall tabula-py with the latest master branch? I haven't released it to PyPI yet.

Suffix should not be the problem since tabula-py automatically adds it.

**Comment by Amorim33** [comment]
Now it is working perfectly, thanks a lot for your help!!

**Comment by chezou** [comment]
Released 2.5.0 https://pypi.org/project/tabula-py/2.5.0/
Thanks for reporting!

**Comment by jgcmarins** [comment]
@chezou thank you!!!
